crates/build/src/docker.rs

```diff
@@ -146,23 +122,43 @@ pub fn docker_build(args: ExecuteArgs) -> Result<BuildResult> {
             let _ = client.ping().await.map_err(|e| {
                 anyhow::anyhow!("{}\nIs your docker engine up and running?", e)
             })?;
-            let build_image =
-                get_image(client.clone(), image, &verbosity, &mut build_steps).await?;
 
-            run_build(
+            let image = match image {
+                ImageVariant::Custom(i) => i.clone(),
+                ImageVariant::Default => {
+                    format!("{}:{}", IMAGE, VERSION)
+                }
+            };
+
+            if let Err(err) =
+                pull_image(client.clone(), &image, &verbosity, &mut build_steps).await
+            {
+                // If the image could not be pulled, we will still attempt to use a local
+                // image of that name if it exists.
+                eprintln!(
```

Comment:
This will always error if the image has never been published. So, that's why I checked if there is a local copy first. Although it might be more code, the error would be more precise.

Reply:
Okay, best of both worlds would be then to optimistically create the container, and if that fails with "not found" then attempt to pull: https://github.com/paritytech/cargo-contract/pull/1210/commits/c7bc50609a15c3fd38f75ca62c3655133f8f18e5